src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/DefaultRemoteTextLoaderFactory.cs

```diff
@@ -72,10 +72,12 @@ public override Task<TextAndVersion> LoadTextAndVersionAsync(Workspace workspace
                         throw new IOException($"File was externally modified: {_filePath}");
                     }
                 }
-                catch (IOException e) when (e is DirectoryNotFoundException || e is FileNotFoundException)
+                catch (IOException)
                 {
                     // This can typically occur when a file is renamed. What happens is the client "closes" the old file before any file system "rename" event makes it to us. Resulting
                     // in us trying to refresh the "closed" files buffer with what's on disk; however, there's nothing actually on disk because the file was renamed.
+                    //
+                    // Can also occur when a file is in the middle of being copied resulting in a generic IO exception for the resource not being ready.
                     textAndVersion = TextAndVersion.Create(SourceText.From(string.Empty), VersionStamp.Default, filePath: _filePath);
```

Comment:
It would be nice/interesting to add a logging statement here so we can start to understand how often we see this (and maybe mitigate if it's a lot) but no need to block the PR on that.

Reply:
Totally thought the same thing; however, because of the re-use of this class this point can't take a logger 😢